internal/support/scheduler/utils.go

```diff
@@ -104,11 +106,12 @@ func parseInt64(value string) int64 {
 // Scheduler Queue Client
 var currentQueueClient *QueueClient // Singleton used so that queueClient can use it to de-reference readings
 type QueueClient struct {
+	loggingClient logger.LoggingClient
 }
 
 // NewClient
-func NewSchedulerQueueClient() *QueueClient {
-	queueClient := &QueueClient{}
+func NewSchedulerQueueClient(loggingClient logger.LoggingClient) *QueueClient {
+	queueClient := &QueueClient{loggingClient}
 	currentQueueClient = queueClient // Set the singleton
 	return queueClient
 }
```

Comment:
Acknowledge you didn't write this code.

It's a mess.  Can you improve it?  Create a separate GitHub issue.  See individual comments.